mllib/src/test/scala/org/apache/spark/ml/regression/LinearRegressionSuite.scala

```diff
@@ -460,6 +460,18 @@ class LinearRegressionSuite extends SparkFunSuite with MLlibTestSparkContext {
     // Training results for the model should be available
     assert(model.hasSummary)
 
+    // Feature column should be equal
+    val expectedFeature = dataset.select("features").collect()
```

Comment:
I don't think there's a need to test the feature column.  I would instead compare the schema of "predictions" with the original schema and make sure it's a superset.